### PR TITLE
[CakePHP] Drop 2.x support in favor of 3.x

### LINKED ISSUE #101: Drop CakePHP 2.x support in favor of 3.x
This issue is here for getting feedback about something I would like to drop. Basically, CakePHP 3.x has been released with namespace support. I would like to migrate the current adapter to this new version but I wouldn't want to maintain two adapters for both versions.

So, is anyone agree about dropping the CakePHP 2.x support in favor of the 3.x one?

### PR DESCRIPTION
**Pull request by egeloen**
This PR fixes #101

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,11 +15,11 @@
         "phly/http": "^0.12.0"
     },
     "require-dev": {
-        "cakephp/cakephp": "^2.5.7",
+        "cakephp/cakephp": "dev-network-http",
         "ext-curl": "*",
         "guzzle/guzzle": "^3.9.2",
-        "guzzlehttp/guzzle": ">=4.1.4,<6.0.0",
-        "guzzlehttp/streams": ">=1.5.0,<4.0.0",
+        "guzzlehttp/guzzle": ">=4.1.4,<6.0",
```
Comment (GrahamCampbell) [inline composer.json]: What about `^4.1.4|~5.0`?

File: composer.json [hunk 2]
```diff
@@ -15,11 +15,11 @@
...
+        "guzzlehttp/streams": ">=1.5,<4.0",
```
Comment (GrahamCampbell) [inline composer.json]: What about `~1.5|~2.0|~3.0`?

File: composer.json (hunk 1 above)
Comment (xabbuh) [inline composer.json]: Must be `~4.1.4|~5.0` then (`^4.1.4` also includes `6.0.0` for example).

Comment (egeloen) [inline composer.json]: @xabbuh AFAIR on https://getcomposer.org/doc/01-basic-usage.md#package-versions, this is not true. `4.1.4` will only allow `4.1.x` version and `^4.1.4` is similar to `>=4.1.4,<5.0`, so IMO, @GrahamCampbell is right here.

Comment (egeloen) [inline composer.json]: Btw, @GrahamCampbell are you sure a single pipe is equivalent to a double pipe? https://getcomposer.org/doc/01-basic-usage.md#package-versions states to use a double pipe in order to make a logical OR.

Comment (xabbuh) [inline composer.json]: @egeloen Sorry, that was indeed wrong.

**Comment by egeloen** [comment]
This PR is in standby since the CakePHP team merge the master branch into the 3.1 one.

**Pull request merged**